### PR TITLE
Fix potential many-to-many errors

### PR DESCRIPTION
**Pull request by simonpasquier**
xref https://bugzilla.redhat.com/show_bug.cgi?id=1802941

### CODE REVIEW

File: rules/node.libsonnet [hunk 1]
```diff
@@ -33,7 +34,7 @@
               count by (%(clusterLabel)s, node) (sum by (node, cpu) (
                 node_cpu_seconds_total{%(nodeExporterSelector)s}
               * on (namespace, %(podLabel)s) group_left(node)
-                node_namespace_pod:kube_pod_info:
+                node_namespace_pod:kube_pod_info:{%(podLabel)s=~"%(nodeExporterPodPrefix)s.+"}
```
Comment (simonpasquier) [inline rules/node.libsonnet]: another option is:

```
               count by (%(clusterLabel)s, node) (sum by (node, cpu) (
                 node_cpu_seconds_total{%(nodeExporterSelector)s}
               * on (namespace, %(podLabel)s) group_left(node)
                 topk by(namespace,pod) (1, node_namespace_pod:kube_pod_info:)
               ))
```

Comment (brancz) [reply rules/node.libsonnet]: I prefer the topk approach

File: rules/node.libsonnet [hunk 2]
```diff
@@ -33,7 +33,7 @@
               count by (%(clusterLabel)s, node) (sum by (node, cpu) (
                 node_cpu_seconds_total{%(nodeExporterSelector)s}
               * on (namespace, %(podLabel)s) group_left(node)
-                node_namespace_pod:kube_pod_info:
+                topk by(namespace,pod) (1, node_namespace_pod:kube_pod_info:)
```
Comment (brancz) [inline rules/node.libsonnet]: Sorry that I didn’t think about this before but why don’t we fix this in the original recording rule here? I would expect this recording rule to always only have one entry per namespace pod combination.

Comment (simonpasquier) [reply rules/node.libsonnet]: Ack, it makes sense.

**Comment by simonpasquier** [comment]
@brancz ready for another review.

**Comment by brancz** [comment]
Thanks! Lgtm!

**Comment by bboreham** [comment]
Can you please clarify how we can get an entry in `kube_pod_info` for one pod on more than one node at the same time?

**Comment by brancz** [comment]
It's an edge case, but in openshift for example where there are often automatic updates of kube-state-metrics, during rollout there may be multiple kube-state-metrics pods being scraped with the same series. This is mostly about making these rules resilient to those cases.

**Pull request merged**